conversion.c

```diff
@@ -0,0 +1,96 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Ayisha Sameera,GET.
+// ------------------------------------------------------------------------------------------------
+// conversion.c
+// Functions to convert a decimal number to binary and hexadecimal.
+// ------------------------------------------------------------------------------------------------
+#pragma warning (disable:4996)
+#include <stdbool.h>
+#include <malloc.h>
+#include <string.h>
+#include <limits.h>
+#include "conversion.h"
+
+/// <summary>Returns the nearest bit length.</summary>
+int NearestBitLen (int digit) {
+	int bitLen = 8;
+	while (bitLen < digit) bitLen <<= 1;
+	return bitLen;
+}
+
+/// <summary>Returns the binary equivalent of input to string.</summary>
+char* BinaryString (bool isNegative, int len, long int num, bool isThirtyTwo) {

```

Comment:
It is generally considered good practice to avoid passing bools as function parameters. See [here](https://www.bing.com/search?q=why+shpuld+we+void+passing+bools+to+functions&cvid=74b7c999d7fc494488413c1011785b7a&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIGCAEQABhAMgYIAhAAGEAyBggDEAAYQDIICAQQ6QcY_FXSAQkxMjY1MmowajGoAgCwAgA&FORM=ANAB01&PC=U531)
```suggestion
char* BinaryString (int len, long int num) {
```
num < 0 and num == 32 work the same way.